padertorch/data/fragmenter.py

```diff
@@ -15,110 +16,199 @@
-    >>> pprint(channel_fragmenter(example))
-    [{'a': array([[0, 1, 2, 3]]), 'b': array([1, 2, 3, 4])},
-     {'a': array([[4, 5, 6, 7]]), 'b': array([1, 2, 3, 4])}]
-    >>> channel_fragmenter = Fragmenter(\
-            {'a':1}, axis=0, squeeze=True, copy_keys=['b'])
-    >>> example = {'a': np.arange(8).reshape((2, 4)), 'b': np.array([1,2,3,4])}
-    >>> pprint(channel_fragmenter(example))
-    [{'a': array([0, 1, 2, 3]), 'b': array([1, 2, 3, 4])},
-     {'a': array([4, 5, 6, 7]), 'b': array([1, 2, 3, 4])}]
+    This segmenting returns a list of segmented examples that can be unbatched.
+    Everything that is not listed in `keys` is simply copied from the
+    input example to the output examples.
+    The keys `segment_start` and `segment_stop` are added for each output
+    dictionary.
+
+    If an utterance is shorter than `length`, a
+    `lazy_dataset.FilterException` is raised.
+
+    Examples (For more examples see tests/test_data/test_segmenter:
+
+        >>> segmenter = Segmenter(length=32000, include_keys=('x', 'y'),\
+                                   shift=16000)
+        >>> ex = {'x': np.arange(65000), 'y': np.arange(65000),\
+                    'num_samples': 65000, 'gender': 'm'}
+        >>> segmented = segmenter(ex)
+        >>> type(segmented)
+        <class 'list'>
+        >>> for entry in segmented:
+        ...     print(entry['x'][[0, -1]])
+        [    0 31999]
+        [16000 47999]
+        [32000 63999]
+
+        segmenting can be disabled by setting `length=-1` or `keys=None`
+        >>> Segmenter(length=-1, include_keys=('x', 'y'))(ex)[0] == ex
+        True
+
+        Check the corner cases.
+        >>> ex = {'x': np.arange(64000), 'y': np.arange(64000)}
+        >>> for entry in segmenter(ex):
+        ...     print(entry['x'][[0, -1]])
+        [    0 31999]
+        [16000 47999]
+        [32000 63999]
+        >>> ex = {'x': np.arange(63999), 'y': np.arange(63999)}
+        >>> for entry in segmenter(ex):
+        ...     print(entry['x'][[0, -1]])
+        [    0 31999]
+        [16000 47999]
+
+    Args:
+        length: The length of the segments in samples. If set to `-1`,
+            the original example is returned in a list of length one.
+        shift: shift between segments, defaults to length
+        include_keys: The keys in the passed example dict to segment. They all
+            must have the same size along their specified `axis`, if keys is
+            None the segmentation is applied to all `numpy.arrays`.
+            If a key points to a dictionary the segmentation is applied to all
+            values of this dictionary
+        exclude_keys: This option allows to specify specific keys not to
+            segment. This might be usefull if `include_keys` is `None` or
+            one of the included keys points to a dictionary.
+        copy_keys: If `None` all not values which are not segmented are
```

Comment:
Default value of `True` sounds more resonable.

Reply:
I thought about that, but I dislike the use of `is True` to check whether it is equal to True this seems more natural with None, but okay, I can change that.